KHLY2400/KHLY2400-IOC-01App/Db/KHLY2400.db

```diff
@@ -585,78 +674,468 @@ alias("$(P)SIM:SOURCE:MODE", "$(P)SIM:SOURCE:MODE:SP:RBV")
+    info(archive, "VAL")
+    info(INTEREST, "LOW")
+}
+
+record(bi, "$(P)SIM:CURR:SOURCE:AUTORANGE")
+{
+}
+
+alias("$(P)CURR:SOURCE:AUTORANGE:SP:RBV", "$(P)CURR:SOURCE:AUTORANGE")
+alias("$(P)SIM:CURR:SOURCE:AUTORANGE", "$(P)SIM:CURR:SOURCE:AUTORANGE:SP")
+alias("$(P)SIM:CURR:SOURCE:AUTORANGE", "$(P)SIM:CURR:SOURCE:AUTORANGE:SP:RBV")
+
+record(bo, "$(P)VOLT:SOURCE:AUTORANGE:SP")
+{
+    field(DESC, "Sets voltage source autoranging on/off")
+    field(DTYP, "stream")
+    field(OUT, "@Keithley2400.proto set_V_SOURCE_AUTORANGE $(PORT)")
+    field(PINI, "YES")
+    field(ZNAM, "Off")
+    field(ONAM, "On")
+    field(VAL, "1")
+    field(SDIS, "$(P)DISABLE")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:VOLT:SOURCE:AUTORANGE:SP")
+    info(archive, "VAL")
+    info(INTEREST, "LOW")
+}
+
+record(bi, "$(P)VOLT:SOURCE:AUTORANGE:SP:RBV")
+{
+    field(DESC, "Gets source voltage autoranging status")
+    field(DTYP, "stream")
+    field(SCAN, "Passive")
+    field(INP, "@Keithley2400.proto get_V_SOURCE_AUTORANGE $(PORT)")
+    field(ZNAM, "Off")
+    field(ONAM, "On")
+    field(SDIS, "$(P)DISABLE")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:VOLT:SOURCE:AUTORANGE")
+    info(archive, "VAL")
+    info(INTEREST, "LOW")
+}
+
+record(bi, "$(P)SIM:VOLT:SOURCE:AUTORANGE")
+{
+}
+
+alias("$(P)VOLT:SOURCE:AUTORANGE:SP:RBV", "$(P)VOLT:SOURCE:AUTORANGE")
+alias("$(P)SIM:VOLT:SOURCE:AUTORANGE", "$(P)SIM:VOLT:SOURCE:AUTORANGE:SP")
+alias("$(P)SIM:VOLT:SOURCE:AUTORANGE", "$(P)SIM:VOLT:SOURCE:AUTORANGE:SP:RBV")
+
+record(ao, "$(P)CURR:SOURCE:RANGE:SP")
+{
+    field(DESC, "Sets the source current range")
+    field(DTYP, "stream")
+    field(OUT, "@Keithley2400.proto set_I_SOURCE_RANGE $(PORT)")
+    field(PREC, "7")
+    field(EGU, "A")
+    field(SDIS, "$(P)DISABLE")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:CURR:SOURCE:RANGE:SP")
+    info(archive, "VAL")
+    info(INTEREST, "LOW")
+}
+
+record(ai, "$(P)CURR:SOURCE:RANGE:SP:RBV")
+{
+    field(DESC, "Gets the source current range")
+    field(DTYP, "stream")
+    field(INP, "@Keithley2400.proto get_I_SOURCE_RANGE $(PORT)")
+    field(PREC, "7")
+    field(EGU, "A")
+    field(SDIS, "$(P)DISABLE")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:CURR:SOURCE:RANGE:SP:RBV")
+    info(archive, "VAL")
+    info(INTEREST, "LOW")
+}
+
+record(ai, "$(P)SIM:CURR:SOURCE:RANGE")
+{
+}
+
+alias("$(P)CURR:SOURCE:RANGE:SP:RBV", "$(P)CURR:SOURCE:RANGE")
+alias("$(P)SIM:CURR:SOURCE:RANGE", "$(P)SIM:CURR:SOURCE:RANGE:SP")
+alias("$(P)SIM:CURR:SOURCE:RANGE", "$(P)SIM:CURR:SOURCE:RANGE:SP:RBV")
+
+record(ao, "$(P)VOLT:SOURCE:RANGE:SP")
+{
+    field(DESC, "Sets the voltage source range")
+    field(DTYP, "stream")
+    field(OUT, "@Keithley2400.proto set_V_SOURCE_RANGE $(PORT)")
+    field(EGU, "V")
+    field(PREC, "10")
+    field(SDIS, "$(P)DISABLE")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:VOLT:SOURCE:RANGE:SP")
+    info(archive, "VAL")
+    info(INTEREST, "LOW")
```

Comment:
Setpoints typically not interesting (put interest on `$(P)THING` as opposed to `$(P)THING:SP` or `$(P)THING:SP:RBV`)

Comment repeats as necessary